services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/taydennys/TaydennysService.kt

```diff
@@ -125,8 +125,17 @@ class TaydennysService(
             "A hakemus has has entered handling. Checking if there's a täydennyspyyntö for the hakemus. ${application.logString()}"
         }
 
-        taydennyspyyntoRepository.findByApplicationId(application.id)?.let {
+        taydennysRepository.findByApplicationId(application.id)?.also {
```

Comment:
This function is called `removeTaydennyspyyntoIfItExists` but it does a little more than that. Also, could the name indicate that this is called only by Allu event handler? I cannot give any example of good name candidates though. Maybe adding some comment for the function is sufficient?

Reply:
What more does it do?

Reply:
It removes täydennys as well.

Reply:
The täydennys is always deleted when the täydennyspyyntö is deleted. The database cascades take care of that. It's done explicitly to log the deletion and delete the attachments.

What makes this limited in use is that it assumes Allu initiated the deletion. Usually, we pass a string for the current user that's used as the actor in the audit logging, but here don't pass anything and log everything to have been done by Allu.

I added a comment to highlight this.